javaslang/src/main/java/javaslang/Value.java

```diff
@@ -1199,6 +1201,12 @@ default CharSeq toCharSeq() {
         return ValueModule.toTraversable(this, Vector.empty(), Vector::of, Vector::ofAll);
     }
 
+    @Override
+    default Spliterator<T> spliterator() {
+        return Spliterators.spliterator(iterator(), isEmpty() ? 0 : 1,
+                Spliterator.IMMUTABLE | Spliterator.DISTINCT | Spliterator.SIZED | Spliterator.SUBSIZED);
```

Comment:
I ask myself if SUBSIZED can be omitted here because there is at most one element in a single-valued type. It cannot be splitted anymore.

But wait... Java 9 added Optional.stream(). I will ask jshell:

```java
jshell> Optional.of(1).stream().spliterator().characteristics() & Spliterator.SUBSIZED
$3 ==> 16384
```

Ok, we need SUBSIZED :)